index.bs

```diff
@@ -225,7 +225,17 @@ its \[[Prototype]] an empty object, created as if by ObjectCreate(%ObjectPrototy
 
 <h4 id="assert" oldids="assert-condition-data,dom-console-assert" method for="console">assert(<var>condition</var>, ...<var>data</var>)</h4>
 
-If <var>condition</var> is false, perform Logger("error", <var>data</var>).
+<emu-alg>
+  1. If _condition_ is true, abort these steps.
+  1. Let _message_ be a string without any formatting specifiers indicating generically an assertion failure (such as "Assertion failed").
+  1. If _data_ is empty, append _message_ to _data_.
+  1. Otherwise, implementations should perform these substeps:
+    1. Let _first_ be the first element of _data_.
+    1. If _first_ is not a string, prepend _message_ to _data_. Abort these substeps.
```

Comment:
Sadly can't phrase it that easily. Only correct statement is: No implementation that concatenates prints `[Object object]`. Only Safari and Chrome have a prefix (and therefore a desire to concat:

- Safari concats and does a type test.
- Chrome doesn't concat (and therefore breaks formatting capabilities).

---

Test cases:
- `console.assert(false, { answer: 42 })`
- `console.assert(false, "foo: %s", "bar")`

Results:

- Chrome
    - `Assertion failed: Object {answer: 42}` (Object is expandable)
    - `Assertion failed: foo: %s bar`
- Safari:
    - `Assertion Failed – {answer: 42}` (Object behind long-dash which is their "rest" divider)
    - `Assertion Failed: foo: bar`
- Firefox:
    - `Object { answer: 42 }` (Object is expandable)
    - `foo: bar`
- Edge:
    - `[Object object]`
        - Note: Edge's `Printer` is a simple `toString` and does not print objects nicely - even for `console.log`
    - `foo: bar`